hail/python/hail/vds/methods.py

```diff
@@ -980,6 +981,10 @@ def truncate_reference_blocks(ds, *, max_ref_block_base_pairs=None, ref_block_wi
     )
     new_rd = new_rd.annotate_globals(**{fd_name: max_ref_block_base_pairs})
 
+    # we've changed LEN so we need to make sure that END is correct.
+    if 'END' in new_rd.entry:
+        new_rd = new_rd.annotate_entries(END=new_rd.LEN + new_rd.locus.position - 1)
```

Comment:
Any chance this can be done in the `es.transmute` expression where `LEN` is redefined so this context can be better contained and not span a few other operations?

Reply:
yep, not a problem!